### DISCUSSION TITLE
Client TLS handshake failed. The client may not trust the proxy's certificate for www.google.com (OpenSSL Error([('SSL routines', 'tls_early_post_process_client_hello', 'inappropriate fallback')]))  server disconnect www.google.com:443

### DISCUSSION THREAD

**Discussion by vs0055** [mitmproxy/mitmproxy#5827] [Q&A]
Hi
I am new to mitmproxy trying this out on my system to intercept https traffic on website like youtube.com or google.com.
but I am facing this issue 

Client TLS handshake failed. The client may not trust the proxy's certificate for www.google.com (OpenSSL Error([('SSL routines', 'tls_early_post_process_client_hello', 'inappropriate fallback')]))  server disconnect www.google.com:443

I have downloaded the certificates and trusted them on my Mac.
but still facing this issue.

Any help is appreciated.

**Comment by zauceee** [comment]
The server itself doesnt approve the certificates I think

**Comment by mhils** [comment]
https://docs.mitmproxy.org/dev/concepts-certificates/#certificate-pinning

**Reply by prog110** [reply]
The app's calls are intercepted by charles proxy properly but not by mitmproxy in my case. Dont know why.

**Reply by fdciabdul** [reply]
mitm proxy are just support TLSv2

**Comment by Stephen0910** [comment]
May I skip TLSv1 requests by any code.
Because lots of hosts should be ignore.
Thanks a lot.

**Reply by nikhilgodfather** [reply]
are you getting any solution because i also facing same error

**Reply by Stephen0910** [reply]
> are you getting any solution because i also facing same error

https://github.com/mitmproxy/mitmproxy/blob/main/examples/contrib/tls_passthrough.py
